objects/src/accounts/account_id.rs

```diff
@@ -40,39 +63,241 @@ pub enum AccountType {
+                AccountPoW(REGULAR_ACCOUNT_MINIMUM_POW)
             },
         }
     }
 }
 
-// ACCOUNT STORAGE TYPES
-// ================================================================================================
+impl Display for AccountType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            AccountType::FungibleFaucet => write!(f, "fungible faucet"),
+            AccountType::NonFungibleFaucet => write!(f, "non-fungible faucet"),
+            AccountType::RegularAccountImmutableCode => write!(f, "immutable regular account"),
+            AccountType::RegularAccountUpdatableCode => write!(f, "updatable regular account"),
+        }
+    }
+}
+
+impl TryFrom<u64> for AccountType {
+    type Error = AccountError;
+
+    fn try_from(value: u64) -> Result<Self, AccountError> {
+        match value {
+            REGULAR_ACCOUNT_UPDATABLE_CODE => Ok(AccountType::RegularAccountUpdatableCode),
+            REGULAR_ACCOUNT_IMMUTABLE_CODE => Ok(AccountType::RegularAccountImmutableCode),
+            FUNGIBLE_FAUCET => Ok(AccountType::FungibleFaucet),
+            NON_FUNGIBLE_FAUCET => Ok(AccountType::NonFungibleFaucet),
+            v => Err(AccountError::InvalidAccountType(v)),
+        }
+    }
+}
+
+impl TryFrom<Felt> for AccountType {
+    type Error = AccountError;
+
+    fn try_from(value: Felt) -> Result<Self, AccountError> {
+        value.as_int().try_into()
+    }
+}
+
+impl From<AccountType> for u64 {
+    fn from(value: AccountType) -> Self {
+        value as u64
+    }
+}
 
-pub const ON_CHAIN: u64 = 0b00;
-pub const OFF_CHAIN: u64 = 0b10;
+// ACCOUNT STORAGE TYPE
+// ================================================================================================
 
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
 #[repr(u64)]
 pub enum AccountStorageType {
-    OnChain = ON_CHAIN,
-    OffChain = OFF_CHAIN,
+    OnChain = ACCOUNT_ON_CHAIN,
+    OffChain = ACCOUNT_OFF_CHAIN,
+}
+
+impl Display for AccountStorageType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            AccountStorageType::OnChain => write!(f, "on-chain"),
+            AccountStorageType::OffChain => write!(f, "off-chain"),
+        }
+    }
+}
+
+impl TryFrom<u64> for AccountStorageType {
+    type Error = AccountError;
+
+    fn try_from(value: u64) -> Result<Self, Self::Error> {
+        match value {
+            ACCOUNT_ON_CHAIN => Ok(AccountStorageType::OnChain),
+            ACCOUNT_OFF_CHAIN => Ok(AccountStorageType::OffChain),
+            v => Err(AccountError::InvalidStorageType(v)),
+        }
+    }
+}
+
+impl TryFrom<Felt> for AccountStorageType {
+    type Error = AccountError;
+
+    fn try_from(value: Felt) -> Result<Self, Self::Error> {
+        value.as_int().try_into()
+    }
+}
+
+impl From<AccountStorageType> for u64 {
```

Comment:
Note: The `u64` conversions from these types expects the bits to:

- be in the right position, w.r.t the account id format
- to represent a valid variant

the above makes it simple to `or` these objects into the final account id, and to extract them by doing a simple bitmask